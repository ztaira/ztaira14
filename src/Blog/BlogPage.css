.BlogContent {
  margin: 80px 2% 50px 2%;
  padding: 0px;
  min-width: 250px;
  width: 96%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.FormattingDiv {
  width: 100%;
  display: flex;
  flex-direction: column;
  flex: 1 0 250px;
}
