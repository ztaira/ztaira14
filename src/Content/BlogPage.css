.BlogContent {
  margin: 80px 4% 50px 4%;
  padding: 0px;
  min-width: 250px;
  width: 92%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.FormattingDiv {
  width: 100%;
  display: flex;
  flex-direction: column;
  flex: 1 0 250px;
}
