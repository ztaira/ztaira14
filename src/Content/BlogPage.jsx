import React, { Component } from 'react';
import BlogItem from './BlogItem.jsx';
import './BlogPage.css';

class BlogPage extends Component {
  constructor(props) {
    super(props);
    this.ReturnBlogItem = this.ReturnBlogItem.bind(this);
  }

  ReturnBlogItem(entry) {
    return <BlogItem key={entry.name} entry={entry} />;
  }

  render() {
    return (
      <div className="BlogContent">
        {this.props.allEntries.map(this.ReturnBlogItem)}
        <div className="RightColumnSpacer" />
      </div>
    );
  }
}

export default BlogPage;
