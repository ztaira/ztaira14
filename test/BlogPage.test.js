import React from 'react';
import ReactDOM from 'react-dom';
import { HashRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import BlogPage from '../src/Blog/BlogPage.jsx';

it('has not changed since the last snapshot', function() {
  const blogpage = renderer.create(
    <HashRouter>
      <BlogPage allEntries={[
        {
          'name': 'Hello Again, world!',
          'description': 'Testing',
          'file_name': 'hello_again_world.md',
          'create_date': '2017-05-29'
        },
        {
          'name': 'Hello World',
          'description': 'Testing',
          'file_name': 'hello_world.md',
          'create_date': '2017-05-29'
        }
      ]} />
    </HashRouter>
  );
  let blogpage_json = blogpage.toJSON();
  expect(blogpage_json).toMatchSnapshot();
});
