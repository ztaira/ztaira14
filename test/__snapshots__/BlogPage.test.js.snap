exports[`test has not changed since the last snapshot 1`] = `
<div
  className="BlogContent">
  <div
    className="FormattingDiv">
    <a
      className="BlogItem"
      onClick={[Function]}>
      <div
        className="BlogItemLeftInfo">
        <div
          className="BlogItemLink">
          hello_again_world
        </div>
        <p
          className="BlogItemDescription">
          Testing
        </p>
      </div>
      <p
        className="BlogItemDate">
        2017-05-29
      </p>
    </a>
    <a
      className="BlogItem"
      onClick={[Function]}>
      <div
        className="BlogItemLeftInfo">
        <div
          className="BlogItemLink">
          hello_world
        </div>
        <p
          className="BlogItemDescription">
          Testing
        </p>
      </div>
      <p
        className="BlogItemDate">
        2017-05-29
      </p>
    </a>
  </div>
  <div
    className="RightColumnSpacer" />
</div>
`;
